controllers/storage.go

```diff
@@ -145,6 +145,34 @@ func (s *Storage) RemoveAllButCurrent(artifact sourcev1.Artifact) ([]string, err
 	return deletedFiles, nil
 }
 
+func (s *Storage) RemoveConditionally(dir string, callbacks []func(path string, info os.FileInfo) bool) ([]string, error) {
```

Comment:
For the callback, let's make this a variadic function and define a new function type for the callback, something like:

```go
type removeCondition func(path string, info os.FileInfo) bool
```
some description would be helpful in the future.
And then change the function signature to:

```go
func (s *Storage) RemoveConditionally(dir string, callbacks ...removeCondition) ([]string, error)
```

With this the caller doesn't have to create a slice, instead just pass an instance of the callback.

Please use of some appropriate name for the type, just giving an example here at the moment.

Reply:
Also, a description of for `RemoveConditionally()` would be nice, public method.